### ISSUE TITLE
Extend Distance Class

### ISSUE THREAD

**Issue by Segelzwerg** [Segelzwerg/SportToolBox#7]
The distance class should be able to handle meters as well. These means the Speed Class must be updated as well.

- [x] add constructor in Distance with allows meters as well
- [ ] overflow method which add meters to kilometer if meter is greater than 1000
- [ ] update Speed class so in handles meters as well
- [x] refactor the primitives from float to int, as there is now no need for decimal values in hour

Tests would be great, but there is a separate issue: #6 If you do both, please reference both issue in PR.

**Comment by retsbol** [comment]
Hi, I'm new to open source and looking for some easy practices. Could I take this one?

**Comment by Segelzwerg** [comment]
yes most definitely. If you need any help, please let me know and don't be afraid of doing something "wrong", we'll figure it out together.

**Comment by retsbol** [comment]
Okay, thank you very much!

**Comment by retsbol** [comment]
I've got some questions:

> overflow method which add meters to kilometer if meter is greater than 1000

Does this refer to a reusable method which can be invoked by a `Distance` instance? Or just need to ensure `meter` field not having an unreasonable value after the initialization in the constructor (without defining extra method)?

I was thinking the first one but then I saw `final` modifier before `kilometer` field. I'm not sure if I should discard `final` and write a method that can reassign the value of `kilometer` for overflow checking or keep `final` and just write the overflow checking logic in the constructor :'(

> update Speed class so in handles meters as well

I found that the computation is actually occurred in `Time` class, so I assume it's a typo. Actually I should update the `Time` class, am I correct?

> refactor the primitives from float to int, as there is now no need for decimal values in hour

Does this simply mean changing the type of `hour` field in `Time` class without modifying any other logic?

Seems to ask a little more... thanks again for your patience!
